src/Uno.UI/UI/Xaml/FrameworkElement.cs

```diff
@@ -438,6 +441,63 @@ private void OnRequestedThemeChanged(ElementTheme oldValue, ElementTheme newValu
 			Application.Current?.ActualElementTheme ?? ElementTheme.Default
 			: ElementTheme.Default;
 
+		[GeneratedDependencyProperty]
+		public static DependencyProperty FocusVisualSecondaryThicknessProperty { get; } = CreateFocusVisualSecondaryThicknessProperty();
+
+		public Thickness FocusVisualSecondaryThickness
+		{
+			get => GetFocusVisualSecondaryThicknessValue();
+			set => SetFocusVisualSecondaryThicknessValue(value);
+		}
+
+		private static Thickness GetFocusVisualSecondaryThicknessDefaultValue() => new Thickness(1);
+
+		[GeneratedDependencyProperty(DefaultValue = default(Brush))]
+		public static DependencyProperty FocusVisualSecondaryBrushProperty { get; } = CreateFocusVisualSecondaryBrushProperty();
+
+		public Brush FocusVisualSecondaryBrush
+		{
+			get => GetFocusVisualSecondaryBrushValue();
+			set => SetFocusVisualSecondaryBrushValue(value);
+		}
+
+		[GeneratedDependencyProperty]
+		public static DependencyProperty FocusVisualPrimaryThicknessProperty { get; } = CreateFocusVisualPrimaryThicknessProperty();
+
+		public Thickness FocusVisualPrimaryThickness
+		{
+			get => GetFocusVisualPrimaryThicknessValue();
+			set => SetFocusVisualPrimaryThicknessValue(value);
+		}
+
+		private static Thickness GetFocusVisualPrimaryThicknessDefaultValue() => new Thickness(2);
+
+		[GeneratedDependencyProperty(DefaultValue = default(Brush))]
+		public static DependencyProperty FocusVisualPrimaryBrushProperty { get; } = CreateFocusVisualPrimaryBrushProperty();
+
+		public Brush FocusVisualPrimaryBrush
+		{
+			get => GetFocusVisualPrimaryBrushValue();
+			set => SetFocusVisualPrimaryBrushValue(value);
+		}
+
+		[GeneratedDependencyProperty]
+		public static DependencyProperty FocusVisualMarginProperty { get; } = CreateFocusVisualMarginProperty();
+
+		public Thickness FocusVisualMargin
+		{
+			get => GetFocusVisualMarginValue();
+			set => SetFocusVisualMarginValue(value);
+		}
+
+		private static Thickness GetFocusVisualMarginDefaultValue() => Thickness.Empty;
+
+		private void InitializeFocusVisualProperties()
+		{
+			ResourceResolver.ApplyResource(this, FocusVisualPrimaryBrushProperty, new SpecializedResourceDictionary.ResourceKey("SystemControlFocusVisualPrimaryBrush"), true, null, DependencyPropertyValuePrecedences.DefaultValue);
```

Comment:
Hmm... this means that every single FrameworkElement in the visual tree is going to be listening for updates to a theme-bound value. There may be perf implications. I wonder if it would be possible to only subscribe for updates if we know that the control is actually being used as a focus visual?

cc @jeromelaban

Reply:
There will be performance implication for sure. Maybe it should be initialized lazily and only if it can be focused ?

Reply:
In that case I would need to be able to "plug-in" that lazy initialization when someone calls `element.GetValue(FocusVisualPrimaryBrushProperty)` - not sure if we have such capability currently 🤔 .

Reply:
@davidjohnoliver @jeromelaban Maybe could use an approach similar to this and just resolve the static resource instead of `ThemeResource`? Static lookup for two resources per element could be quite cheap
https://github.com/unoplatform/uno/blob/56d3ba98ee6cab055dc8d7c4941ce4494e20e946/src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs#L931-L935?

Reply:
@MartinZikmund Sounds like a good way to go - static resource look-up isn't a problem, only the theme-update binding is a potential concern.

Reply:
Though the more I think about it, I wonder if we still want to avoid the static look-up for every single element - it still has to walk upward and check all available resources.

Reply:
Yes, it is possible to override the value be declaring a resource with the same name unfortunately 🤔

Reply:
@davidjohnoliver  @jeromelaban  Adjusted code to use static retrieval

Reply:
If this would still be too performance intensive, I could make the default to return some hardcoded color (like transparent) and in the property getter initialize the DP lazily. The only disadvantage would then be that `GetValue(FocusVisualPrimaryBrushProperty)` will return wrong value until `FocusVisualPrimaryBrush` is accessed. To make sure it works for the focused element, the `SystemFocusVisual` control could force this initialization when the data-bound element changes

Reply:
It's definitely better, but I wonder if even the static walk up the tree is too costly if it's done for every single element.

Actually, since you're doing this from the constructor, it's never going to find elements in the visual tree (eg `Page.Resources`), only application-level resources. 

So I wonder if this can be statically cached, so you only have to do it once, the first time it's requested? (If the theme changes, you invalidate the cache.)

Reply:
Yes, I will do this lazy initialization instead, I think it is acceptable solution